code/weapon/weapons.cpp

```diff
@@ -872,10 +872,24 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		first_time = true;
 
 		strcpy_s(wip->name, fname);
+
+		// if this name has a hash, create a default display name
+		if (get_pointer_to_first_hash_symbol(wip->name)) {
```

Comment:
Same as for ships probably applies here as well.

Reply:
See above.